src/assets.rs

```diff
@@ -289,13 +288,17 @@ impl HighlightingAssets {
                 None => {
                     let file_name = path.file_name().unwrap_or_default();
                     self.get_extension_syntax(file_name)?
-                        .or(line_syntax)
                         .ok_or_else(|| Error::UndetectedSyntax(path.to_string_lossy().into()))
                 }
             }
         } else {
             // If a path wasn't provided, we fall back to the detect first-line syntax.
-            line_syntax.ok_or_else(|| Error::UndetectedSyntax("[unknown]".into()))
+            Err(Error::UndetectedSyntax("[unknown]".into()))
+        };
+
+        match path_syntax {
+            Err(err) => self.get_first_line_syntax(&mut input.reader)?.ok_or(err),
```

Comment:
maybe we should show some type of warning that the mapped syntax was invalid/ignored?

Reply:
Hmm yes maybe it is confusing to change the behavior as I did after all.

Rather than creating new problems, I will aim for not changing behavior from before, while still keeping the simplification of the code. I will soon push a commit with code that restores old behavior but keeps the refactoring, so we can discuss it.